scan-command/tool-scan/README.md

```diff
@@ -115,4 +115,666 @@ bal scan --platforms=sonarqube
+
+#### Noncompliant Code Example
+
+```ballerina
+public function checkResult() {
+    json result = checkpanic getResult();
+
+    // ...
+}
+
+public function getResult() returns json|error {
+    // ...
+}
+```
+
+#### Compliant Solution
+
+Check and handle the error explicitly.
+
+```ballerina
+public function checkResult() {
+    json|error result = getResult();
+
+    if result is error {
+        // handle error
+    }
+
+    // ...
+}
+
+public function getResult() returns json|error {
+    // ...
+}
+```
+
+Make use of the 'check' keyword, which immediately returns the error or transfers control to an on-fail block, 
+in contrast to 'checkpanic' and panicking if an expression or action evaluates to an error.
+
+```ballerina
+public function checkResult() returns error? {
+    json result = check getResult();
+}
+
+public function getResult() returns json|error {
+    // ...
+}
+```
+
+### ballerina:2 - Unused function parameter
+
+Unused function parameters cause unnecessary code complexity and can lead to confusion for developers
+maintaining the code. They may also indicate potential errors in function design or changes in requirements
+that were not properly implemented.
+
+#### Noncompliant Code Example
+
+```ballerina
+import ballerina/io;
+
+function name(int a, int b) {
+   io:println(a);
+}
+
+public function main() {
+   name(1, 2);
+}
+```
+
+#### Compliant Solution
+
+Remove unused function parameters to improve code clarity and maintainability. If the parameter is intended for 
+future use or completeness, consider documenting it or refactoring the function to use it.
+
+```ballerina
+import ballerina/io;
+
+function name(int a) {
+   io:println(a);
+}
+
+public function main() {
+   name(1);
+}
+```
+
+### ballerina:3 - Unused function parameter
+A non-isolated function will not be called concurrently. Only isolated functions are called concurrently given that
+they are guaranteed to be safe if the arguments are also safe. To allow being called concurrently,
+a public function should be marked as isolated.
+
+#### Noncompliant Code Example
+
+```ballerina
+public function helperFunction() {
+   // isolated code
+}
+```
+
+#### Compliant Solution
+Mark public functions as isolated to ensure the function can be called concurrently.
+
+```ballerina
+public isolated function helperFunction() {
+   // isolated code
+}
+```
+
+### ballerina:4 - Non isolated public method
+
+Class methods can be isolated. An isolated method is the same as an isolated function with self treated as a parameter.
+A non-isolated method will not be called concurrently. Only isolated methods are called concurrently given that
+they are guaranteed to be safe if the arguments are also safe. To allow being called concurrently,
+a public method should be marked as isolated.
+
+#### Noncompliant Code Example
+
+```ballerina
+class EvenNumber {
+int i = 1;
+
+    public function generate() returns int {
+        lock {
```

Comment:
Shall we remove this lock in non-compliant and add this to compliant sample only?